persistence/src/commonMain/kotlin/com/wire/kalium/persistence/dao/UserDAOImpl.kt

```diff
@@ -77,18 +77,38 @@ class UserDAOImpl(
     override suspend fun upsertUsers(users: List<UserEntity>) {
         userQueries.transaction {
             for (user: UserEntity in users) {
-                userQueries.updateUser(
-                    user.name,
-                    user.handle,
-                    user.email,
-                    user.phone,
-                    user.accentId,
-                    user.team,
-                    user.previewAssetId,
-                    user.completeAssetId,
-                    user.userTypEntity,
-                    user.id,
-                )
+                /**
+                 * We have only access to [UserTypeEntity] from our team
+                 * by fetching team members
+                 * So we need to avoid updating that field
+                 * when team member is fetched from different endpoint
+                 */
+                if (user.userTypeEntity == UserTypeEntity.INTERNAL) {
+                    userQueries.updateUserWithoutType(
+                        user.name,
+                        user.handle,
+                        user.email,
+                        user.phone,
+                        user.accentId,
+                        user.team,
+                        user.previewAssetId,
+                        user.completeAssetId,
+                        user.id,
+                    )
+                } else {
+                    userQueries.updateUser(
+                        user.name,
+                        user.handle,
+                        user.email,
+                        user.phone,
+                        user.accentId,
+                        user.team,
+                        user.previewAssetId,
+                        user.completeAssetId,
+                        user.userTypeEntity,
+                        user.id,
+                    )
+                }
```

Comment:
I'm not sure I like having this kind of logic in the DAO.

DAO should be a stupid data storage. Seems that this could be moved at least to the Repository, that knows about different sources and can care about combining data from multiple sources.


At the very least, a couple of test cases covering this behaviour would be very welcome.

Reply:
Moved logic to repository, I just need to write tests for them if you agree with current solution 😉